### ISSUE TITLE
[Crash] IncrementalDelayCalculator

### ISSUE THREAD

**Issue by rlinoz** [qonversion/android-sdk#172]
We are having the following crash, it looks like it happens when coming back from background.

```
Fatal Exception: java.lang.IllegalArgumentException
bound must be positive
java.util.Random.nextInt (Random.java:388)
com.qonversion.android.sdk.IncrementalDelayCalculator.countDelay (IncrementalDelayCalculator.java:19)
com.qonversion.android.sdk.QUserPropertiesManager$forceSendProperties$2.invoke (QUserPropertiesManager.java:70)
com.qonversion.android.sdk.QUserPropertiesManager$forceSendProperties$2.invoke (QUserPropertiesManager.java:11)
com.qonversion.android.sdk.QonversionRepository$sendProperties$1$2.invoke (QonversionRepository.java:101)
com.qonversion.android.sdk.QonversionRepository$sendProperties$1$2.invoke (QonversionRepository.java:23)
com.qonversion.android.sdk.CallBackKt.onFailure (CallBackKt.java:20)
retrofit2.DefaultCallAdapterFactory$ExecutorCallbackCall$1.lambda$onFailure$1 (DefaultCallAdapterFactory.java:96)
retrofit2.DefaultCallAdapterFactory$ExecutorCallbackCall$1.lambda$onFailure$1$DefaultCallAdapterFactory$ExecutorCallbackCall$1 (DefaultCallAdapterFactory.java:1)
```

Platform: Android
SDK Version: 2.8.1

**Comment by Maria-Bordunova** [comment]
Hi, @rlinoz 
Thank you for the report. It's really weird as there is nowhere to take a negative value. 
We couldn't reproduce the scenario. Please share some more details:
- How often do you encounter the error? 
- Maybe you have details about the users who got this error 
Any information can be useful

**Comment by rlinoz** [comment]
Hi @Maria-Bordunova 

It has happened 78 times for 68 users in the last 7 days, right now it is the crash with the highest rate we have.

Devices data:
<img src="https://user-images.githubusercontent.com/5201282/125648426-513732cc-514e-40f1-8ebe-91d7009319fc.png" width="400" />

For every crash, it was the last event reported by each user, so it may be associated with some uninstall process.

**Comment by Maria-Bordunova** [comment]
Hi again @rlinoz 
We eventually found the bug. It is produced only after **five and a half hours** of the app work. The push notifications in your app launch it in background mode and trigger the sending of some qonversion requests which weren't successful. Therefore SDK retried them. 

The screenshot also confirms this: 92% devices state is background. This crash didn't impact on your users as they did not see the crash :pray:
Anyway, we are sorry for the inconvenience. We fixed the bug and released new [sdk version 2.10.1 ](https://github.com/qonversion/android-sdk/releases/tag/2.10.1).

P.S.Qonversion recently started to support Google Play Billing Library 4.0.0 version and released the third major version of the sdk. The [sdk version 3.0.1](https://github.com/qonversion/android-sdk/releases/tag/3.0.1) also contains the fix.
